src/wallet/wallet.cpp

```diff
@@ -4494,6 +4494,22 @@ void CWallet::LoadActiveScriptPubKeyMan(uint256 id, OutputType type, bool intern
     NotifyCanGetAddressesChanged();
 }
 
+void CWallet::DeactivateScriptPubKeyMan(uint256 id, OutputType type, bool internal)
+{
+    auto& spk_mans = internal ? m_internal_spk_managers : m_external_spk_managers;
+    if (spk_mans[type] && spk_mans[type]->GetID() == id) {
```

Comment:
I think using `[]` for access here could be problematic if the `type` isn't already in the map (I think that can happen). I would prefer using `find` here and then if it works, checking that the returned object is not `nullptr`.

Reply:
I'm not a cpp expert, but the docs say in case when there is no such key a default value will be inserted. 
> std::map<Key,T,Compare,Allocator>::operator[]
> Returns a reference to the value that is mapped to a key equivalent to key, performing an insertion if such key does not already exist.
So I believe it safe in this particular case.

I still prefer `spk_mans[type]` over `spk_mans.find(type) != spk_mans.end() && spk_mans[type] != nullptr` but if that's going against the norms for the project I can change to whatever is considered standard way.

Reply:
Right, but the point is that we don't want a default value inserted.

Reply:
Done